### DISCUSSION TITLE
Component creation. Why doesn't work?

### DISCUSSION THREAD

**Discussion by astatonn** [joomla/joomla-cms#42809] [Q&A]
Hello everyone,

I'm trying to develop my own component, however, I'm encountering difficulties in understanding how to create it.
I have already read the FAQ on component creation, the J4.x component creation article, searched the internet, asked Gemini and ChatGPT, but I can't make progress. :(

My project structure is as follows:

```
administrator/components/com_aniversariantes/services/provider.php
administrator/components/com_aniversariantes/sql/install.mysql.utf8.sql
administrator/components/com_aniversariantes/src/Controller/DisplayController.php
administrator/components/com_aniversariantes/src/view/HtmlView.php
administrator/components/com_aniversariantes/tmpl/aniversariantes/default.php
administrator/components/com_aniversariantes/aniversariantes.xml
components/com_aniversariantes/src/Controller/DisplayController.php
components/com_aniversariantes/view/Aniversariantes/HtmlView.php
components/com_aniversariantes/tmpl/aniversariantes/default.php
```

At this moment, I can display static data in a datatable on my component's admin panel.

What do I want?
I need to code a CRUD for the table that I created and send data to the component to display it on the site section.

I don't want someone to develop my component. I just want to fix the problems I'm encountering and understand how to complete my task.

I would really appreciate it if someone could spare their time to show me a way to solve this because I don't have anyone to do it for me at the moment.

administrator/components/com_aniversariantes/src/Controller/DisplayController.php
```
<?php

namespace Astatonn\Component\Aniversariantes\Administrator\Controller;

defined('_JEXEC') or die;

use Joomla\CMS\MVC\Controller\BaseController;

/**
 * @package     Joomla.Administrator
 * @subpackage  com_aniversariantes
 *
 * @copyright   Copyright (C) 2024 Astatonn. All rights reserved.
 * @license     GNU General Public License version 3; see LICENSE
 */

/**
 * Default Controller of Aniversariantes component
 *
 * @package     Joomla.Administrator
 * @subpackage  com_aniversariantes
 */
class DisplayController extends BaseController {
    /**
     * The default view for the display method.
     *
     * @var string
     */
    protected $default_view = 'aniversariantes';
    
    public function display($cachable = false, $urlparams = array()) {
        return parent::display($cachable, $urlparams);
    }
    
}
```

administrator/components/com_aniversariantes/src/view/HtmlView.php
```
<?php

namespace Astatonn\Component\Aniversariantes\Administrator\View\Aniversariantes;

defined('_JEXEC') or die;

use Joomla\CMS\MVC\View\HtmlView as BaseHtmlView;

/**
 * @package     Joomla.Administrator
 * @subpackage  com_aniversariantes
 *
 * @copyright   Copyright (C) 2024 Lucas de Souza Silva Lima. All rights reserved.
 * @license     GNU General Public License version 3; see LICENSE
 */

/**
 * Main "Aniversariantes" Admin View
 */
class HtmlView extends BaseHtmlView {
    
    /**
     * Display the main "Aniversariantes" view
     *
     * @param   string  $tpl  The name of the template file to parse; automatically searches through the template paths.
     * @return  void
     */
    function display($tpl = null) {
        parent::display($tpl);
    }

}
```

administrator/components/com_aniversariantes/tmpl/aniversariantes/default.php
```
<?php

/**
 * @package     Joomla.Administrator
 * @subpackage  com_aniversariantes
 *
 * @copyright   Copyright (C) 2024 Astatonn. All rights reserved.
 * @license     GNU General Public License version 3; see LICENSE
 */

// No direct access to this file
defined('_JEXEC') or die('Restricted Access');
// use Joomla\CMS\Factory;
// use Joomla\CMS\Categories\Categories;



// $db = Factory::getContainer()->get(DatabaseInterface::class);



?>

<div class="container">
    <div class="title">
        <h1>Aniversariantes</h1>
    </div>

    <section>
        <div class="btn">Adicionar Novo</div>
    </section>

    <div class="table-aniversariantes  margin-table">
        <table class="table" id="tabela">
            <thead>
                <tr>
                    <th>P/G</th>
                    <th>NOME</th>
                    <th>ANIVERSÁRIO</th>
                    <th>AÇÕES</th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td>Row 1 Data 1</td>
                    <td>Row 1 Data 2</td>
                    <td>Row 1 Data 3</td>
                    <td>Row 1 Data 4</td>
                </tr>
                <tr>
                    <td>Row 2 Data 1</td>
                    <td>Row 2 Data 2</td>
                    <td>Row 2 Data 3</td>
                    <td>Row 2 Data 4</td>
                </tr>
            </tbody>
        </table>
    </div>

    <script src="//cdn.datatables.net/1.13.7/js/jquery.dataTables.min.js"> </script>

    <link rel="stylesheet" href="//cdn.datatables.net/1.13.7/css/jquery.dataTables.min.css">
    <script>
        $(document).ready(function() {
            let table = new DataTable('#tabela');
        });
    </script>
</div>
```

**Comment by MarkRS-UK** [comment]
@astatonn , Good for you for starting component development, but you have more research to do.

The J! documentation will take you further on that you currently are.  There's a [J4! basic document](https://docs.joomla.org/J4.x:Developing_an_MVC_Component/Introduction)as well as Robbie's excellent in-depth [J3 article with J4 additions](https://docs.joomla.org/J3.x:Developing_an_MVC_Component).

In short, you need to use a model.

**Comment by chmst** [comment]
Please be aware that a tutorial for J3 is not at all suitable for J4/5. 
Newer documentation for developers will be published here https://manual.joomla.org/

**Reply by MrSuddenJoy** [reply]
@chmst am also interested in developing modules, thats why Im asking. When will the docs be published?
